app/containers/ImageViewer/ImageCarousal.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import { StyleSheet, View } from 'react-native';
+import { GestureHandlerRootView } from 'react-native-gesture-handler';
+import Animated, { runOnJS, useAnimatedReaction, useAnimatedStyle, useSharedValue } from 'react-native-reanimated';
+
+import { ImageViewer } from './ImageViewer';
+
+export interface ImageProps {
+	imageComponentType?: string;
+	width: number;
+	height: number;
+	onLoadEnd?: () => void;
+}
+
+export interface IImageData {
+	uri: string;
+	title?: string;
+	image_type?: string;
+}
+
+interface ImageCarousalProps extends ImageProps {
+	data: IImageData[];
+	firstIndex: number;
+	showHeader?: boolean;
+}
+
+export const ImageCarousal = ({
```

Comment:
We should be using https://github.com/dohooo/react-native-reanimated-carousel

Reply:
@diegolmello Sir, the problem is that when we place it inside another component, we can't pan properly when zoomed because the outer view takes control of the gestures. That's why I added custom pan logic for the image viewer to pan when zooming and optimised it to prevent lag with large data.

Reply:
https://github.com/RocketChat/Rocket.Chat.ReactNative/assets/61021881/b3da1343-8a6b-48a0-967d-0b6a320366e2

Reply:
@diegolmello  I attempted using the reanimated carousel, but when I tried to pan while zooming, it didn't work. Is there a solution for this? Thanks